### PR TITLE
Bug fix for azure connection and get disk by name

### PR DESCRIPTION
**Pull request by qinpingli**
@duanwei33  @pruan-rht Please help review.

logs:
        And I verify that the IAAS volume with id "<%= cb.volumeID %>" was deleted                        # features/step_definitions/pv.rb:187
      [10:03:32] INFO> Shell Commands: oc get infrastructures.config.openshift.io cluster --output=yaml --kubeconfig=/root/workdir/Tester-OCP-qinping/ocp4_admin.kubeconfig
     <--skip-->
      
      [10:03:36] INFO> Exit Status: 0
      [10:03:36] INFO> Shell Commands: oc get secrets azure-credentials --output=yaml --kubeconfig=/root/workdir/Tester-OCP-qinping/ocp4_admin.kubeconfig --namespace=kube-system
      <--skip-->
      
      [10:03:38] INFO> Exit Status: 0
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
waiting for operation up to 3600 seconds..
    [10:03:42] INFO> === After Scenario: dynamic provisioning, Examples (#4) ===

### CODE REVIEW

**Comment by duanwei33** [comment]
/lgtm
Thanks a lot for the update.

File: lib/iaas/iaas.rb [hunk 1]
```diff
@@ -69,8 +69,18 @@ def self.init_gcp(env)
...
+      auth = {}
+      auth[:tenant_id] = secret.value_of("azure_tenant_id", user: :admin)
+      auth[:client_id] = secret.value_of("azure_client_id", user: :admin)
+      auth[:client_secret] = secret.value_of("azure_client_secret", user: :admin)
+
+      return BushSlicer::Azure.new(
+        :location => secret.value_of("azure_region", user: :admin),
+        :subscription_id => secret.value_of("azure_subscription_id", user: :admin),
+	:auth => auth,
+	:resource_group => secret.value_of("azure_resourcegroup", user: :admin)
```
Comment (liangxia) [inline lib/iaas/iaas.rb:82]: Indention issue.

File: lib/launchers/azure.rb [hunk 2]
```diff
@@ -211,7 +211,16 @@ def get_volume_by_openshift_metadata(pv_name, project_name)
     # @return [TODO, nil] returns nil when not found
     # @raise on communication error
     def get_volume_by_id(id)
-      TODO
+      name = id.split("/")[-1]
+      res = "exist"
+      begin
+      compute_client.disks.get(azure_config[:resource_group], name)
+      rescue MsRestAzure::AzureOperationError => e
+	if e.response.status == 404
```
Comment (liangxia) [inline lib/launchers/azure.rb:219]: Indention issue.

File: lib/iaas/iaas.rb (hunk 1 above)
Comment (qinpingli) [reply lib/iaas/iaas.rb]: Thanks for the review, updated.

File: lib/launchers/azure.rb (hunk 2 above)
Comment (qinpingli) [reply lib/launchers/azure.rb]: done

**Pull request merged**